README.md

```diff
@@ -25,162 +33,111 @@ implementation "com.careem.mockingbird:mockingbird:$mockingBirdVersion"
+```
 
-    object Method {
-        const val method1 = "method1"
-        const val method2 = "method2"
-        const val method3 = "method3"
-        const val method4 = "method4"
-    }
+Groovy DSL:
 
-    object Arg {
-        const val str = "str"
-        const val value = "value"
-        const val value1 = "value1"
-        const val value2 = "value2"
-    }
+```groovy
+apply plugin: "com.careem.mockingbird"
+```
 
-    override fun method1(str: String) = spy(
-        methodName = Method.method1,
-        arguments = mapOf(
-            Arg.str to str
-        ),
-        delegate = { delegate.method1(str) }
-    )
+And then specify for what interfaces you what to generate mocks for, see the example below
 
-    override fun method2(str: String, value: Int) = spy(
-        methodName = Method.method2,
-        arguments = mapOf(
-            Arg.str to str,
-            Arg.value to value
-        ),
-        delegate = { delegate.method2(str, value) }
-    )
+Kotlin DSL:
 
-    override fun method3(value1: Int, value2: Int): Int = spy(
-        methodName = Method.method3,
-        arguments = mapOf(
-            Arg.value1 to value1,
-            Arg.value2 to value2
-        ),
-        delegate = { delegate.method3(value1, value2) }
-    )
-
-    override fun method4(): Int = spy(
-        methodName = Method.method4,
-        delegate = { delegate.method4() }
+```kotlin
+configure<com.careem.mockingbird.MockingbirdPluginExtension> {
+    generateMocksFor = listOf(
+        "com.careem.mockingbird.sample.Mock1",
+        "com.careem.mockingbird.sample.MockWithExternalDependencies",
+        "com.careem.mockingbird.common.sample.ExternalContract"
     )
 }
+```
 
-class MyDependencyImpl : MyDependency {
-    private var value: AtomicInt = atomic(0)
-    override fun method1(str: String) {
+Groovy DSL
 
-    }
+```groovy
+mockingBird {
+    generateMocksFor = [
+            'com.careem.mockingbird.sample.Mock1',
+            'com.careem.mockingbird.sample.MockWithExternalDependencies',
+            'com.careem.mockingbird.common.sample.ExternalContract'
+    ]
+}
+```
 
-    override fun method2(str: String, value: Int) {
+#### Plugin Usage
 
-    }
+To generate mocks you can simply run `./gradlew generateMocks` or simply run `./gradlew build`, for faster development
+loop using `generatedMocks` is recommended
 
-    override fun method3(value1: Int, value2: Int): Int {
-        value.value = value1 + value2
-        return value.value
-    }
+#### Plugin Limitations
 
-    override fun method4(): Int {
-        return value.value
-    }
-}
-```
+* The plugin can only be used in modules containing a `jvm` target
+* The plugin can generate mocks only, no support for spies yet
+* You can mock interfaces only
+* You cannot mock interfaces that have generic types in their definitions
+* You cannot mock lambdas
+* You cannot mock suspend functions
+* You cannot mock inline functions
+* You cannot mock reified functions
```

Comment:
Nit: I'd suggest to make these passive, like `Only interfaces can be mocked`, `Generic interfaces cannot be mocked` etc.